logger/src/main/java/com/orhanobut/logger/FileLogger.java

```diff
@@ -0,0 +1,362 @@
+package com.orhanobut.logger;
+
+import android.os.Environment;
+import android.os.Handler;
+import android.os.HandlerThread;
+import android.os.Looper;
+import android.os.Message;
+import android.util.Log;
+import android.util.SparseArray;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+import java.util.Locale;
+
+import static com.orhanobut.logger.LoggerPrinter.BOTTOM_BORDER;
+import static com.orhanobut.logger.LoggerPrinter.HORIZONTAL_DOUBLE_LINE;
+import static com.orhanobut.logger.LoggerPrinter.MIDDLE_BORDER;
+import static com.orhanobut.logger.LoggerPrinter.TOP_BORDER;
+
+/**
+ * Created by ronaldo pace on 1/5/17. https://github.com/budius
+ * <p>
+ * Typical implementation for the file Logger could look like:
+ * <pre>
+ * {@code
+ * Logger.init().logAdapter(
+ *         new MultiLogAdapter(Arrays.asList(
+ *         new AndroidLogAdapter(),  // standard android logcat
+ *         FileLogger.builder()      // file logger
+ *             .folderName("myApp")  // folder (starting from Environment.getExternalStorageDirectory())
+ *             .logLevel(android.util.Log.INFO) // only log INFO and up
+ *             .build())));
+ * }
+ * </pre>
+ * Alternatively <code>.folderFullPath(path)</code> can be used to log on internal memory
+ */
+public class FileLogger implements LogAdapter {
+
+    //region static definitions
+    private static final String BEGIN = TOP_BORDER;
+    private static final String MID = MIDDLE_BORDER;
+    private static final String END = BOTTOM_BORDER;
+    private static final String HORZ_LINE = String.valueOf(HORIZONTAL_DOUBLE_LINE);
+    private static final String CSV_SEPARATOR = ",";
+    private static final String NEW_LINE = System.getProperty("line.separator");
+    private static final String FILE_SEPARATOR = File.separator;
+    // I'm using br here because it's easy to find search, and won't be confused easily
+    private static final String NEW_LINE_REPLACEMENT = " <br> ";
+    private static final int DEFAULT_MAX_LINES = 2048;
+    private static final List<String> MARKERS = Arrays.asList(BEGIN, MID, END);
+
+    private static final SparseArray<String> LEVELS;
+
+    static {
+        LEVELS = new SparseArray<>();
+        LEVELS.append(Log.VERBOSE, "VERBOSE");
+        LEVELS.append(Log.DEBUG, "DEBUG");
+        LEVELS.append(Log.INFO, "INFO");
+        LEVELS.append(Log.WARN, "WARN");
+        LEVELS.append(Log.ERROR, "ERROR");
+        LEVELS.append(Log.ASSERT, "ASSERT");
+    }
+    //endregion
+
+    private final Handler handler;
+    private final StringBuilder stringBuilder = new StringBuilder(200);
+    private final SimpleDateFormat dateFormat;
+    private final Date date = new Date();
+    private final String separator;
+    private final int logLevel;
```

Comment:
This is where stuff starts diverging a bit.
As common logs uses VERBOSE, DEBUG, INFO, WARNING, ERROR, it's an inherently useful use case to write to disk only from a certain level and up, as for a offline checking, one might not need VERBOSE or even DEBUG, but only INFO (for example).

I understand adding a second `logType` to the library as whole generates confusion, but having a FileLogger always log from VERBOSE and up adds a lot of noise to possible useful data.

I will suggest here to keep FULL (which will be the same as verbose) and add all the other common levels. But that wouldn't remove the need to have a separate value for the disk logger, it would simply unify the values used across the lib.